Added extra task for Squeaky Clean exercise

trippshelnutt opened the pull request:
I was working on the Squeaky Clean exercise and it wasn't explicitly clear to me that letters should be removed. I added an extra task and instruction to clarify the requirements.

ErikSchierboom reviewed (commented):
Thanks for the PR! The instruction do state:

> A valid identifier comprises zero or more letters and underscores.

but I agree that this might be too subtle and a new test is a good idea. On small tweak: could you make this the fourth task and the existing fourth task the fifth? 

I just looked at the [test](https://github.com/exercism/csharp/blob/main/exercises/concept/squeaky-clean/SqueakyCleanTests.cs), and I feel like they could use a fair bit of attention too. For one, the order of the tests and the task to which they're linked doesn't fully match the instructions.
Would you be willing to also update the tests at the same time?

trippshelnutt commented:
No problem! I reordered the tasks in the instructions and moved some of the tests around to match the tasks. I also added a test explicitly to test for numbers.

The last test `Combine_conversions` seems to combine elements from all tasks. I left it associated with the last task. Do you have any feedback as to how the `Task` attribute should account for this?

ErikSchierboom reviewed (approved):
> The last test Combine_conversions seems to combine elements from all tasks. I left it associated with the last task. Do you have any feedback as to how the Task attribute should account for this?

That's perfectly fine!
